src/core/src/sim/mc/moves/resize.rs

```diff
@@ -0,0 +1,112 @@
+// Cymbalum, an extensible molecular simulation engine
+// Copyright (C) 2015-2016 G. Fraux — BSD license
+
+use rand::distributions::{Sample, Range};
+use rand::Rng;
+
+use std::f64;
+use std::mem;
+
+use super::MCMove;
+
+use types::{Matrix3, One};
+use sys::{System, EnergyCache};
+
+/// Monte-Carlo move that changes the size of the simulation cell
+pub struct Resize {
+    /// Delta for translation of the box length
+    delta: f64,
+    /// Sampling range for volume scaling
+    range: Range<f64>,
+    /// System before applying changes to the simulation box
+    old_system: System,
+    /// target pressure
+    pressure: f64,
+    /// largest cut off diameter of `PairPotentials`
+    rc_max: f64,
+}
+
+impl Resize {
+    /// Create a new `Resize` move, with target pressure `pressure` and maximum
+    /// displacement of `delta`.
+    pub fn new(pressure: f64, delta: f64) -> Resize {
+        assert!(delta > 0.0, "delta must be positive in Resize move");
+        // TODO: set rc_max from the largest cut off in PairPotentials
+        Resize {
+            delta: 0.,
+            range: Range::new(-delta, delta),
+            old_system: System::new(),
+            pressure: pressure,
+            rc_max: 100.0, // arbitrarily large value
+        }
+    }
+}
+
+impl MCMove for Resize {
+    fn describe(&self) -> &str {
+        "resizing of the cell"
+    }
+
+    fn prepare(&mut self, system: &mut System, rng: &mut Box<Rng>) -> bool {
+        self.delta = self.range.sample(rng);
+        // clone the current (old) system
+        self.old_system = system.clone();
+        let volume = system.volume();
+        let scaling_factor = f64::cbrt((volume + self.delta) / volume);
+        // change the simulation cell
+        system.cell_mut().scale_mut(Matrix3::one() * scaling_factor);
+        let new_cell = system.cell().clone();
+        // check the radius of the smallest inscribed sphere and compare to the 
+        // cut off distance.
+        // abort simulation when box gets too small
+        if new_cell.lengths_as_vec().iter().any(|&d| 0.5 * d < self.rc_max) {
+            fatal_error!(
```

Comment:
Well, wrong energies lead to wrong configurations. In the end, the ensemble averages computed are incorrect. We could adjust the cutoff so that it fits into the new box. That is done in some codes. Personally, I think this solution is also not a very good idea, since it essentially changes the force field.